website/docs/r/s3_bucket_lifecycle_configuration.html.markdown

```diff
@@ -367,6 +367,7 @@ This resource supports the following arguments:
 * `bucket` - (Required) Name of the source S3 bucket you want Amazon S3 to monitor.
 * `expected_bucket_owner` - (Optional) Account ID of the expected bucket owner. If the bucket is owned by a different account, the request will fail with an HTTP 403 (Access Denied) error.
 * `rule` - (Required) List of configuration blocks describing the rules managing the replication. [See below](#rule).
+* `transition_default_minimum_object_size` - (Optional) Which default minimum object size behavior is applied to the lifecycle configuration. Valid values: `all_storage_classes_128K`, `varies_by_storage_class`. To customize the minimum object size for any transition you can add a `filter` that specifies a custom `object_size_greater_than` or `object_size_less_than` value. Custom filters always take precedence over the default transition behavior.
```

Comment:
Should we also add an example with a custom filter to achieve the previous default behavior for the storage classes impacted by this change?

We may also want a changelog note describing how updates to existing configurations may cause behavioral changes since the default transition limit will change when a lifecycle configuration is updated.

Reply:
I'll incorporate https://github.com/hashicorp/terraform-provider-aws/pull/39492 also.

Reply:
I modified the [_Specifying a filter based on object size_ example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_lifecycle_configuration#specifying-a-filter-based-on-object-size) to match the relevant [Amazon S3 User Guide example](https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-configuration-examples.html#lc-small-objects).